src/write_buffer_manager.rs

```diff
@@ -0,0 +1,188 @@
+///! `WriteBufferManager` is for managing memory allocation for one or more
+///! MemTables.
+use std::ptr::NonNull;
+use std::sync::Arc;
+
+use crate::{ffi, Cache};
+
+pub(crate) struct WriteBufferManagerWrapper {
+    pub(crate) inner: NonNull<ffi::rocksdb_write_buffer_manager_t>,
+}
+
+// Just like the types in `db_options`, these are only safe because
+// the underlying types are thread-safe. This thread-safety is not actually
+// documented, but an evaluation of the code revealed that all member
+// variables are protected by mutexes or atomics.
```

Comment:
What is `db_options` ? Can you give a link?

Also, I think it's a bit dubious to claim that "every member variable is wrapped in a mutex" is equivalent to "the object is thread-safe".

Reply:
I can clarify this as best as I can